packages/core/src/lib/store/index.ts

```diff
@@ -72,361 +67,104 @@ export interface QueryOptions {
-   */
-  async queryWithOrderedCallback<T extends IDecodedMessage>(
-    decoders: IDecoder<T>[],
-    callback: (message: T) => Promise<void | boolean> | boolean | void,
-    options?: QueryOptions
-  ): Promise<void> {
-    for await (const promises of this.queryGenerator(decoders, options)) {
-      if (await this.processMessages(promises, callback, options)) break;
-    }
-  }
-
-  /**
-   * Do a query to a Waku Store to retrieve historical/missed messages.
-   * The callback function takes a `Promise<WakuMessage>` in input,
-   * useful if messages need to be decrypted and performance matters.
-   *
-   * The order of the messages passed to the callback is as follows:
-   * - within a page, messages are expected to be ordered from oldest to most recent
-   * - pages direction depends on { @link QueryOptions.pageDirection }
-   *
-   * Do note that the resolution of the `Promise<WakuMessage | undefined` may
-   * break the order as it may rely on the browser decryption API, which in turn,
-   * may have a different speed depending on the type of decryption.
-   *
-   * @throws If not able to reach a Waku Store peer to query,
-   * or if an error is encountered when processing the reply,
-   * or if two decoders with the same content topic are passed.
-   */
-  async queryWithPromiseCallback<T extends IDecodedMessage>(
-    decoders: IDecoder<T>[],
-    callback: (
-      message: Promise<T | undefined>
-    ) => Promise<void | boolean> | boolean | void,
-    options?: QueryOptions
-  ): Promise<void> {
-    let abort = false;
-    for await (const page of this.queryGenerator(decoders, options)) {
-      const _promises = page.map(async (msgPromise) => {
-        if (abort) return;
-        abort = Boolean(await callback(msgPromise));
-      });
-
-      await Promise.all(_promises);
-      if (abort) break;
-    }
-  }
-
-  /**
-   * Do a query to a Waku Store to retrieve historical/missed messages.
-   *
-   * This is a generator, useful if you want most control on how messages
-   * are processed.
-   *
-   * The order of the messages returned by the remote Waku node SHOULD BE
-   * as follows:
-   * - within a page, messages SHOULD be ordered from oldest to most recent
-   * - pages direction depends on { @link QueryOptions.pageDirection }
-   * @throws If not able to reach a Waku Store peer to query,
-   * or if an error is encountered when processing the reply,
-   * or if two decoders with the same content topic are passed.
-   *
-   * This API only supports querying a single pubsub topic at a time.
-   * If multiple decoders are provided, they must all have the same pubsub topic.
-   * @throws If multiple decoders with different pubsub topics are provided.
-   * @throws If no decoders are provided.
-   * @throws If no decoders are found for the provided pubsub topic.
-   */
-  async *queryGenerator<T extends IDecodedMessage>(
-    decoders: IDecoder<T>[],
-    options?: QueryOptions
+  async *queryPerPage<T extends IDecodedMessage>(
```

Comment:
isn't is just `queryPage` or `queryMessages`?

Reply:
correct. it's just `query`, that is a generator that queries per page.

i didn't change the function name to keep things similar to the previous code, but we can definitely change this to just `query` in the future